### PR TITLE
Fix failures unit test php8.1 part2

### PR DESCRIPTION
**Pull request by karyna-t**
### Description (*)
This PR is fixing existing Unit Tests (and related files from Magento Core) to be compatible with PHP 8.1.

### Related Issues
1. magento/magento2#34441

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by karyna-tsymbal-atwix** [comment]
@magento run all tests

**Comment by karyna-tsymbal-atwix** [comment]
@magento run all tests

File: app/code/Magento/Customer/Controller/Adminhtml/Index/InlineEdit.php [hunk 1]
```diff
@@ -179,7 +179,7 @@ protected function getData(array $data, $isCustomerData = null)
         $addressKeys = preg_grep(
             '/^(' . AttributeRepository::BILLING_ADDRESS_PREFIX . '\w+)/',
             array_keys($data),
-            $isCustomerData
+            $isCustomerData ?? 0
```
Comment (andrewbess) [inline app/code/Magento/Customer/Controller/Adminhtml/Index/InlineEdit.php]: Hello @karyna-tsymbal-atwix 
The $isCustomerData is mixed param, however `preg_grep` function requires integer there.
Perhaps, we need to cast it.
```suggestion
            (int) $isCustomerData ?: 0
```

or

```suggestion
            is_int($isCustomerData) ? $isCustomerData : 0
```
What do you think about it?
cc @xmav

**Review by andrewbess — CHANGES_REQUESTED** [review]
Hello @karyna-tsymbal-atwix 
Thank you for your contribution.
Please check my comments.
Thank you in advance.

File: app/code/Magento/Customer/Controller/Adminhtml/Index/InlineEdit.php (hunk 1 above)
Comment (xmav) [reply app/code/Magento/Customer/Controller/Adminhtml/Index/InlineEdit.php]: `(int) $isCustomerData` should cover current argument value variations:[null, true] and other possible variations(if any)

Comment (karyna-t) [reply app/code/Magento/Customer/Controller/Adminhtml/Index/InlineEdit.php]: I replaced it with (int) cast

**Comment by karyna-tsymbal-atwix** [comment]
@magento run all tests

**Review by andrewbess — APPROVED** [review]
Hello @karyna-tsymbal-atwix 
Thank you for your contribution
Approved ✔️

**Comment by xmav** [comment]
pending delivery https://github.com/magento-commerce/magento2ce/pull/7295

**Pull request merged**